### DISCUSSION TITLE
OCR giving out Random Characters

### DISCUSSION THREAD

**Discussion by Tsamsiyu1** [ocrmypdf/OCRmyPDF#1410] [Q&A]
I have a couple documents (PDF's) that already have readable text. When i put them into paperless (it uses OCRmyPDF) and the OCR finishes it is not readable anymore.
This is a screenshot of part of the Original pdf.
![image](https://github.com/user-attachments/assets/b4343d22-329f-4849-9fee-15fe9223d5b7)

and this is the screenshot of the same part but of the archive file:
![image](https://github.com/user-attachments/assets/8cf4580e-ff0d-4f03-9413-feae4bb0d82d)

my OCR is set to german and english and worked fine till now. all the other documents are processed correctly. just those 4 files look like this.
Is this a setting I have wrong or is the file itself protected?

**Comment by ke-ma-fi** [comment]
I am encountering an issue with invoices from a specific firm. While processing 4,000 documents in my system, I didn’t experience this problem with any other documents. However, invoices from this particular firm consistently display the same random characters, as previously described.

**Comment by bernd3k** [comment]
This has the feeling of a bad atttempt at an obfuscation layer à la ROT13 within the PDF - though it's not "shift by 13" in this case. Every symbol is replaced by the same other symbol in the entire document.

| original char| ASCII code | replacement char | ASCII code | shift |
| :----:|-------------:| :-----: |-------------:|-----------:|
|  A | 65 | $ | 39 | 26 |
| b |98 | E | 69 | 29 |
| w |119 | Z | 90|29 |
| a | 97 | D| 68 | 29|
| s | 115 | V| 86 | 29 |
| s | 115 | V| 86| 29 |
| e | 101 | H | 72| 29 |
| r | 114 | U| 85| 29 |

The A is shifted less because they might need the logical replacement "!" for something else. All numbers must have been shifted into the realm of control characters (unprintable) or filtered out entirely.

**Reply by jbarlow83** [reply]
When PDFs do not have a Unicode mapping table, each PDF reader has its own strategy to guess the Unicode mapping. Some guess poorly.

Also, some Unicode mapping tables are corrupt or incorrect, and then again, each PDF reader has its own strategy.

**Comment by luziferius** [comment]
There's a very similar case that came up in another discussion in Paperless-ngx, this time with a solution: According to https://github.com/paperless-ngx/paperless-ngx/discussions/8861#discussioncomment-12955954, the cause is missing embedded fonts. Running another tool to convert the offending document to PDF/A solved the issue.
